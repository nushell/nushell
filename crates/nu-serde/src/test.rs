use crate::{to_success_return_values, to_value};
use insta::assert_debug_snapshot;
use nu_source::Tag;
use serde::Serialize;
use std::collections::BTreeMap;

#[test]
fn it_works_with_single_integers() {
    assert_debug_snapshot!(to_value(&4i32, Tag::default()));
}

#[test]
fn it_works_with_lists_of_values() {
    assert_debug_snapshot!(to_value(&vec![4i32, 10, 8843234, 100], Tag::default()));
}

#[test]
fn it_works_with_complex_structs() {
    #[derive(Serialize, Debug)]
    struct Complex {
        index: i64,
        x: f32,
        y: f64,
        map: BTreeMap<String, Vec<u8>>,
    }

    let mut map = BTreeMap::new();

    map.insert("coconuts".into(), vec![4]);
    map.insert("tilapia".into(), vec![16, 3, 24]);
    map.insert("mahi mahi".into(), vec![]);

    assert_debug_snapshot!(to_value(
        &Complex {
            index: -40,
            x: 32.8,
            y: 38.2,
            map
        },
        Tag::default()
    ));
}

#[test]
fn it_serializes_return_value_list() {
    assert_debug_snapshot!(to_success_return_values(
        vec![4i32, 10, 8843234, 100],
        Tag::default()
    ));
}
