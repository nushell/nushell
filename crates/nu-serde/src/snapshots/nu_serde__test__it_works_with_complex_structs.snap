---
source: src/test.rs
expression: "to_value(&Complex{index: -40, x: 32.8, y: 38.2, map,})"

---
Ok(
    Value {
        value: Row(
            Dictionary {
                entries: {
                    "index": Value {
                        value: Primitive(
                            Int(
                                -40,
                            ),
                        ),
                        tag: Tag {
                            anchor: None,
                            span: Span {
                                start: 0,
                                end: 0,
                            },
                        },
                    },
                    "x": Value {
                        value: Primitive(
                            Decimal(
                                BigDecimal("32.80000"),
                            ),
                        ),
                        tag: Tag {
                            anchor: None,
                            span: Span {
                                start: 0,
                                end: 0,
                            },
                        },
                    },
                    "y": Value {
                        value: Primitive(
                            Decimal(
                                BigDecimal("38.20000000000000"),
                            ),
                        ),
                        tag: Tag {
                            anchor: None,
                            span: Span {
                                start: 0,
                                end: 0,
                            },
                        },
                    },
                    "map": Value {
                        value: Row(
                            Dictionary {
                                entries: {
                                    "coconuts": Value {
                                        value: Table(
                                            [
                                                Value {
                                                    value: Primitive(
                                                        Int(
                                                            4,
                                                        ),
                                                    ),
                                                    tag: Tag {
                                                        anchor: None,
                                                        span: Span {
                                                            start: 0,
                                                            end: 0,
                                                        },
                                                    },
                                                },
                                            ],
                                        ),
                                        tag: Tag {
                                            anchor: None,
                                            span: Span {
                                                start: 0,
                                                end: 0,
                                            },
                                        },
                                    },
                                    "mahi mahi": Value {
                                        value: Table(
                                            [],
                                        ),
                                        tag: Tag {
                                            anchor: None,
                                            span: Span {
                                                start: 0,
                                                end: 0,
                                            },
                                        },
                                    },
                                    "tilapia": Value {
                                        value: Table(
                                            [
                                                Value {
                                                    value: Primitive(
                                                        Int(
                                                            16,
                                                        ),
                                                    ),
                                                    tag: Tag {
                                                        anchor: None,
                                                        span: Span {
                                                            start: 0,
                                                            end: 0,
                                                        },
                                                    },
                                                },
                                                Value {
                                                    value: Primitive(
                                                        Int(
                                                            3,
                                                        ),
                                                    ),
                                                    tag: Tag {
                                                        anchor: None,
                                                        span: Span {
                                                            start: 0,
                                                            end: 0,
                                                        },
                                                    },
                                                },
                                                Value {
                                                    value: Primitive(
                                                        Int(
                                                            24,
                                                        ),
                                                    ),
                                                    tag: Tag {
                                                        anchor: None,
                                                        span: Span {
                                                            start: 0,
                                                            end: 0,
                                                        },
                                                    },
                                                },
                                            ],
                                        ),
                                        tag: Tag {
                                            anchor: None,
                                            span: Span {
                                                start: 0,
                                                end: 0,
                                            },
                                        },
                                    },
                                },
                            },
                        ),
                        tag: Tag {
                            anchor: None,
                            span: Span {
                                start: 0,
                                end: 0,
                            },
                        },
                    },
                },
            },
        ),
        tag: Tag {
            anchor: None,
            span: Span {
                start: 0,
                end: 0,
            },
        },
    },
)
